### PR TITLE
[32880] Display scheduling mode in work package view (full, split & table)

### PR DESCRIPTION
**Pull request by HDinger**
### ToDo

**WP view**
- [x] In the display mode of a date field:
  - [x] There is an icon in front of the start date that indicates the selected scheduling mode.
  - [x] Start and end date of a Work package should be displayed in one `DisplayField`
- [x] In the edit mode of a date field:
  - [x] A local modal with a datepicker and further buttons is shown. 
  - [x] The datepicker is extended/changed to be able to handle start and end date at the same time.
  - [x] You can set the scheduling mode in the date picker modal.
  - [x] Saving the  modal updates all three fields automatically (start date, end date, scheduling icon)
  - [x] For milestones, only one date can be set
  - [x] For Date CustomFields the normal datepicker is shown
- [x] Add/extend tests for new functionality

**Changes to the datepicker**
- [x] The newly introduced datepicker should be used for all datepickers in the application
  - [x] Update the `OpDatePickerComponent` for Angular views
  - [x] Update the `AugmentedDatePicker` for rails views
  - [x] Remove left-overs of the jQuery UI date picker 

**Known bugs**
- [x] When opening the create form, the datepicker is already shown 
- [x] Opening the modal causes a scrollbar to appear
- [x] Saving the modal toggles the scheduling mode without that I made any changes to it
- [x] Going back in the browser, does not close the modal (when going back to angular view)
- [x] Cancelling the WP create form, results in an error "Subject can't be blank"
- [x] Entering dates manually in modal does not always update the datepicker
- [x] Datepicker of CustomFields do not close on focus loss
- [ ] In the WP table (tbd whether we want to fix that):
  - [ ] Due Date for Milestones shows the icon, too.
  - [ ] The edit field shows "startDate - dueDate" although I only opened for example the startDate.

### CODE REVIEW

**Comment by HDinger** [comment]
Tickets:
https://community.openproject.com/projects/openproject/work_packages/32880/activity
https://community.openproject.com/projects/openproject/work_packages/32879/activity

File: frontend/src/app/modules/fields/edit/field-types/date-edit-field.component.ts [hunk 1]
```diff
@@ -26,19 +26,16 @@
...
 // ++
 
-import {Component} from "@angular/core";
+import {Component, OnInit} from "@angular/core";
 import * as moment from "moment";
 import {TimezoneService} from "core-components/datetime/timezone.service";
 import {EditFieldComponent} from "core-app/modules/fields/edit/edit-field.component";
 import {InjectField} from "core-app/helpers/angular/inject-field.decorator";
+import {DatePickerModal} from "core-components/datepicker/datepicker.modal";
+import {OpModalService} from "core-components/op-modals/op-modal.service";
```
Comment (openprojectci) [inline frontend/src/app/modules/fields/edit/field-types/date-edit-field.component.ts:36]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/8ce9d25d81c89fa5502fed98ea5e9dbaa1292905/frontend/src/app/modules/fields/edit/field-types/date-edit-field.component.ts#L36"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/common/op-date-picker/op-date-picker.component.ts [hunk 2]
```diff
@@ -26,45 +26,63 @@
 // See docs/COPYRIGHT.rdoc for more details.
 // ++
 
-import {Component, ElementRef, EventEmitter, Input, OnDestroy, OnInit, Output} from '@angular/core';
+import {AfterViewInit, Component, ElementRef, EventEmitter, Input, OnDestroy, OnInit, Output} from '@angular/core';
 import {ConfigurationService} from 'core-app/modules/common/config/configuration.service';
-import {DatePicker} from 'core-app/modules/common/op-date-picker/datepicker';
 import {TimezoneService} from 'core-components/datetime/timezone.service';
+import flatpickr from "flatpickr";
+import {Instance} from "flatpickr/dist/types/instance";
```
Comment (openprojectci) [inline frontend/src/app/modules/common/op-date-picker/op-date-picker.component.ts:34]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/c95e30f5a1bf5f63b0edaf05f52c85abac58ba63/frontend/src/app/modules/common/op-date-picker/op-date-picker.component.ts#L34"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/common/op-date-picker/op-date-picker.component.ts [hunk 3]
```diff
@@ -26,20 +26,29 @@
 // See docs/COPYRIGHT.rdoc for more details.
 // ++
 
-import {Component, ElementRef, EventEmitter, Input, OnDestroy, OnInit, Output} from '@angular/core';
+import {AfterViewInit, Component, ElementRef, EventEmitter, Input, OnDestroy, OnInit, Output} from '@angular/core';
 import {ConfigurationService} from 'core-app/modules/common/config/configuration.service';
-import {DatePicker} from 'core-app/modules/common/op-date-picker/datepicker';
 import {TimezoneService} from 'core-components/datetime/timezone.service';
+import {DatePicker} from "core-app/modules/common/op-date-picker/datepicker";
```
Comment (openprojectci) [inline frontend/src/app/modules/common/op-date-picker/op-date-picker.component.ts:33]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/77e7ac9b02d9572adf45821544b9e7270c13f581/frontend/src/app/modules/common/op-date-picker/op-date-picker.component.ts#L33"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/fields/edit/field-types/date-edit-field.component.ts [hunk 4]
```diff
@@ -26,36 +26,38 @@
 // See docs/COPYRIGHT.rdoc for more details.
 // ++
 
-import {Component} from "@angular/core";
+import {Component, OnInit} from "@angular/core";
 import * as moment from "moment";
 import {TimezoneService} from "core-components/datetime/timezone.service";
 import {EditFieldComponent} from "core-app/modules/fields/edit/edit-field.component";
 import {InjectField} from "core-app/helpers/angular/inject-field.decorator";
+import {OpModalService} from "core-components/op-modals/op-modal.service";
```
Comment (openprojectci) [inline frontend/src/app/modules/fields/edit/field-types/date-edit-field.component.ts:35]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/99d9a1a8761217071328b21b6c1e03c5349c746a/frontend/src/app/modules/fields/edit/field-types/date-edit-field.component.ts#L35"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

**Comment by openprojectci** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/99d9a1a8761217071328b21b6c1e03c5349c746a/frontend/src/app/components/wp-form-group/wp-attribute-group.component.ts#L34">frontend/src/app/components/wp-form-group/wp-attribute-group.component.ts#L34</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: frontend/src/app/modules/common/op-date-picker/op-date-picker.component.ts [hunk 5]
```diff
@@ -26,73 +26,83 @@
...
 
-import {Component, ElementRef, EventEmitter, Input, OnDestroy, OnInit, Output} from '@angular/core';
+import {AfterViewInit, Component, ElementRef, EventEmitter, Input, OnDestroy, Output, ViewChild} from '@angular/core';
 import {ConfigurationService} from 'core-app/modules/common/config/configuration.service';
-import {DatePicker} from 'core-app/modules/common/op-date-picker/datepicker';
 import {TimezoneService} from 'core-components/datetime/timezone.service';
+import {DatePicker} from "core-app/modules/common/op-date-picker/datepicker";
+import {DebouncedEventEmitter} from "core-components/angular/debounced-event-emitter";
+import {UntilDestroyedMixin} from "core-app/helpers/angular/until-destroyed.mixin";
+import {componentDestroyed} from "@w11k/ngx-componentdestroyed";
```
Comment (openprojectci) [inline frontend/src/app/modules/common/op-date-picker/op-date-picker.component.ts:36]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/f18c489c8863563c25f931526d4af5f3f4f50c6e/frontend/src/app/modules/common/op-date-picker/op-date-picker.component.ts#L36"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/common/op-date-picker/op-date-picker.component.ts [hunk 6]
```diff
@@ -26,93 +26,125 @@
...
+import {keyCodes} from "core-app/modules/common/keyCodes.enum";
+import {Instance} from "flatpickr/dist/types/instance";
```
Comment (openprojectci) [inline frontend/src/app/modules/common/op-date-picker/op-date-picker.component.ts:38]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/99d9a1a8761217071328b21b6c1e03c5349c746a/frontend/src/app/modules/common/op-date-picker/op-date-picker.component.ts#L38"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

**Pull request merged**